Applications/TextEditor/TextEditorWidget.cpp

```diff
@@ -114,8 +114,22 @@ TextEditorWidget::TextEditorWidget()
         statusbar->set_text(builder.to_string());
     };
 
-    m_new_action = GAction::create("New", { Mod_Ctrl, Key_N }, GraphicsBitmap::load_from_file("/res/icons/16x16/new.png"), [](const GAction&) {
-        dbgprintf("FIXME: Implement File/New\n");
+    m_new_action = GAction::create("New", { Mod_Ctrl, Key_N }, GraphicsBitmap::load_from_file("/res/icons/16x16/new.png"), [this](const GAction&) {
+        if (m_document_dirty) {
+            GMessageBox save_document_first_box("Save Document First?", "Warning", GMessageBox::Type::Warning, GMessageBox::InputType::OKCancel, window()); 
+            auto save_document_first_result = save_document_first_box.exec();
+
+            if (save_document_first_result == GDialog::ExecResult::ExecOK) {
```

Comment:
Let's favor "early return" style here and avoid the `else` entirely:

```cpp
if (save_document_first_result == GDialog::ExecResult::ExecCancel)
    return;
```

Reply:
Fixed, It actually has to be 
```
if (save_document_first_result != GDialog::ExecResult::ExecOK) 
    return;
```

because of the possibility of GDialog::ExecResult::ExecAborted.